### PR TITLE
Add Kafka authentication and authorization description in TiCDC

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Add Kafka authentication and authorization description about sasl-gssapi.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/9633
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/manage-ticdc.md [hunk 1]
```diff
@@ -218,12 +218,21 @@ The following are descriptions of parameters and parameter values that can be co
...
-| `sasl-user` | The identity (authcid) of SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
-| `sasl-password` | The password of SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
-| `sasl-mechanism` | The name of SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional)  |
+| `sasl-user` | The identity (authcid) of SASL/PLAIN or SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
+| `sasl-password` | The password of SASL/PLAIN or SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
+| `sasl-mechanism` | The name of SASL/PLAIN or SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional)  |
+| `sasl-gssapi-auth-type` | The gssapi authentication type. Values can be `user` or `keytab` (optional) |
+| `sasl-gssapi-keytab-path` | The gssapi keytab path (optional)|
+| `sasl-gssapi-kerberos-config-path` | The gssapi kerberos configuration path (optional) |
+| `sasl-gssapi-service-name` | The gssapi service name（ (optional) |
```
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:231]:
```suggestion
| `sasl-gssapi-service-name` | The gssapi service name (optional) |
```

File: ticdc/manage-ticdc.md [hunk 2]
```diff
@@ -218,12 +218,21 @@ The following are descriptions of parameters and parameter values that can be co
...
+| `sasl-gssapi-user` | The user name of the gssapi `user` authentication type |
+| `sasl-gssapi-password` |  The password of the gssapi `user` authentication type |
```
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:233]:
```suggestion
| `sasl-gssapi-password` |  The password of gssapi  authentication (optional) |
```

File: ticdc/manage-ticdc.md [hunk 3]
```diff
@@ -238,6 +247,57 @@ Best practices:
...
+  ```
+
+  Values of `sasl-gssapi-user` and `sasl-gssapi-realm` are related to the [principle](https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html) specified in kerberos. For example, if the principle is set as `alice/for-kafka@example.com`, then `sasl-gssapi-user` and `sasl-gssapi-realm` are specified as `alice/for-kafka` and `example.com` respectively.
+
+  SASL/GSSAPI `keytab` authentication type:
+
+  {{< copyable "shell-regular" >}}
+
+  ```shell
+  --sink-uri="kafka://127.0.0.1:9092/topic-name?kafka-version=2.4.0&sasl-mechanism=gssapi&sasl-gssapi-auth-type=keytab&sasl-gssapi-kerberos-config-path=/etc/krb5.conf&sasl-gssapi-service-name=a&sasl-gssapi-user=alice/for-kafka&sasl-gssapi-keytab-path=/var/lib/secret/alice.key&sasl-gssapi-realm=example.com"
```
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:285]:
```suggestion
  --sink-uri="kafka://127.0.0.1:9092/topic-name?kafka-version=2.4.0&sasl-mechanism=gssapi&sasl-gssapi-auth-type=keytab&sasl-gssapi-kerberos-config-path=/etc/krb5.conf&sasl-gssapi-service-name=kafka&sasl-gssapi-user=alice/for-kafka&sasl-gssapi-keytab-path=/var/lib/secret/alice.key&sasl-gssapi-realm=example.com"
```

File: ticdc/manage-ticdc.md [hunk 4]
```diff
@@ -238,6 +247,57 @@ Best practices:
...
+  ```
+
+  For more information about SASL/GSSAPI authentication methods, see [Configuring GSSAPI](https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_gssapi.html).
+
+- TLS/SSL encryption
+
+  If the Kafka broker has TLS/SSL encryption enabled, you need to add the `-enable-tls=true` parameter to `--sink-uri`. If you want to use self-signed certificates, you also need to specify `-ca`, `cert` and `key` in `-sink-uri`.
```
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:292]:
```suggestion
  If the Kafka broker has TLS/SSL encryption enabled, you need to add the `-enable-tls=true` parameter to `--sink-uri`. If you want to use self-signed certificates, you also need to specify `-ca`, `cert` and `key` in `--sink-uri`.
```

Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:292]:
```suggestion
  If the Kafka broker has TLS/SSL encryption enabled, you need to add the `-enable-tls=true` parameter to `--sink-uri`. If you want to use self-signed certificates, you also need to specify `ca`, `cert` and `key` in `--sink-uri`.
```

File: ticdc/manage-ticdc.md [hunk 5]
```diff
@@ -238,6 +247,57 @@ Best practices:
...
+
+- ACL authorization
+
+  The minimum set of permissions required for TiCDC to function properly is as follows.
+
+    - The `Create` and `Write` permissions for the Topic [resources](https://docs.confluent.io/platform/current/kafka/authorization.html#resources).
+    - The `DescribeConfigs` permissions for the Cluster resources.
```
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:299]:
```suggestion
    - The `DescribeConfigs` permission for the Cluster resource type.
```

File: ticdc/manage-ticdc.md [hunk 6]
```diff
@@ -238,6 +247,57 @@ Best practices:
...
+
+- TLS/SSL encryption
+
+  If the Kafka broker has TLS/SSL encryption enabled, you need to add the `-enable-tls=true` parameter to `--sink-uri`. If you want to use self-signed certificates, you also need to specify `-ca`, `cert` and `key` in `-sink-uri`.
+
+- ACL authorization
+
+  The minimum set of permissions required for TiCDC to function properly is as follows.
+
+    - The `Create` and `Write` permissions for the Topic [resources](https://docs.confluent.io/platform/current/kafka/authorization.html#resources).
```
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:298]:
```suggestion
    - The `Create` and `Write` permissions for the Topic [resource type](https://docs.confluent.io/platform/current/kafka/authorization.html#resources).
```

File: ticdc/manage-ticdc.md (hunk 2 above)
Comment (zhaoxinyu) [inline ticdc/manage-ticdc.md:233]:
```suggestion
| `sasl-gssapi-password` | The password of gssapi authentication (optional)  |
```

File: ticdc/manage-ticdc.md [hunk 7]
```diff
@@ -218,12 +218,21 @@ The following are descriptions of parameters and parameter values that can be co
...
 | `ca` | The path of the CA certificate file needed to connect to the downstream Kafka instance (optional)  |
 | `cert` | The path of the certificate file needed to connect to the downstream Kafka instance (optional) |
 | `key` | The path of the certificate key file needed to connect to the downstream Kafka instance (optional) |
-| `sasl-user` | The identity (authcid) of SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
-| `sasl-password` | The password of SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
-| `sasl-mechanism` | The name of SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional)  |
+| `sasl-user` | The identity (authcid) of SASL/PLAIN or SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
+| `sasl-password` | The password of SASL/PLAIN or SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional) |
+| `sasl-mechanism` | The name of SASL/PLAIN or SASL/SCRAM authentication needed to connect to the downstream Kafka instance (optional)  |
```
Comment (ran-huang) [inline ticdc/manage-ticdc.md:227]:
```suggestion
| `sasl-mechanism` | The name of SASL authentication needed to connect to the downstream Kafka instance. The value can be `plain`, `scram-sha-256`, `scram-sha-512`, or `gssapi`. |
```

File: ticdc/manage-ticdc.md [hunk 8]
```diff
@@ -238,6 +247,57 @@ Best practices:
 >
 > When `protocol` is `open-protocol`, TiCDC tries to avoid generating messages that exceed `max-message-bytes` in length. However, if a row is so large that a single change alone exceeds `max-message-bytes` in length, to avoid silent failure, TiCDC tries to output this message and prints a warning in the log.
 
+#### TiCDC uses the authentication and authorization of Kafka
+
+The following are examples when using Kafka SASL authentication:
+
+{{< copyable "shell-regular" >}}
+
```
Comment (ran-huang) [inline ticdc/manage-ticdc.md]:
```suggestion
```

File: ticdc/manage-ticdc.md [hunk 9]
```diff
@@ -238,6 +247,57 @@ Best practices:
 >
 > When `protocol` is `open-protocol`, TiCDC tries to avoid generating messages that exceed `max-message-bytes` in length. However, if a row is so large that a single change alone exceeds `max-message-bytes` in length, to avoid silent failure, TiCDC tries to output this message and prints a warning in the log.
 
+#### TiCDC uses the authentication and authorization of Kafka
+
+The following are examples when using Kafka SASL authentication:
+
+{{< copyable "shell-regular" >}}
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:254]:
```suggestion
```

File: ticdc/manage-ticdc.md [hunk 10]
```diff
@@ -238,6 +247,55 @@ Best practices:
...
+  --sink-uri="kafka://127.0.0.1:9092/topic-name?kafka-version=2.4.0&sasl-user=alice-user&sasl-password=alice-secret&sasl-mechanism=plain"
+  ```
+
+- SASL/SCRAM
+
+  SCRAM-SHA-256 and SCRAM-SHA-512 are similar to the PLAIN method. You just need to specify `sasl-mechanism` as the corresponding authentication method.
+
+- SASL/GSSAPI
+
+  SASL/GSSAPI `user` authentication:
```
Comment (ran-huang) [inline ticdc/manage-ticdc.md:268]:
```suggestion
  SASL/GSSAPI `user` type authentication:
```

File: ticdc/manage-ticdc.md [hunk 11]
```diff
@@ -238,6 +247,55 @@ Best practices:
...
+
+  {{< copyable "shell-regular" >}}
+
+  ```shell
+  --sink-uri="kafka://127.0.0.1:9092/topic-name?kafka-version=2.4.0&sasl-mechanism=gssapi&sasl-gssapi-auth-type=user&sasl-gssapi-kerberos-config-path=/etc/krb5.conf&sasl-gssapi-service-name=kafka&sasl-gssapi-user=alice/for-kafka&sasl-gssapi-password=alice-secret&sasl-gssapi-realm=example.com"
+  ```
+
+  Values of `sasl-gssapi-user` and `sasl-gssapi-realm` are related to the [principle](https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html) specified in kerberos. For example, if the principle is set as `alice/for-kafka@example.com`, then `sasl-gssapi-user` and `sasl-gssapi-realm` are specified as `alice/for-kafka` and `example.com` respectively.
+
+  SASL/GSSAPI `keytab` authentication type:
```
Comment (ran-huang) [inline ticdc/manage-ticdc.md:278]:
```suggestion
  SASL/GSSAPI `keytab` type authentication:
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

**Pull request merged**